### DISCUSSION TITLE
Support WebDav as music provider

### DISCUSSION THREAD

**Discussion by ronaldvandervegt** [music-assistant/support#526] [Feature requests and ideas]
I'm using Nextcloud as my file storage, including media. 
Nextcloud has the possibity to access files using WebDav.

Can we in general get Webdav support?

**Comment by Selim042** [comment]
I'd love for support for NextCloud!!

**Reply by OzGav** [reply]
This post https://community.home-assistant.io/t/music-assistant-2-0-your-music-your-players/727626/564 advises that Nextcloud supports the Subsonic API which you can use with MA

**Comment by Adam-Dalloul** [comment]
Have we gotten any updates on this yet?

**Reply by OzGav** [reply]
No sorry. Work continues on removing the final bugs for a 2.0 release. Then enhancements can be started. The existing music provider poll will guide development effort unless a new dev wants to work in something in particular

**Comment by jem24h** [comment]
I would also love to have support for Support WebDav as music provider. I'm using pcloud (the only cloud with a lifetime price / no subscription) and it supports WebDav. Ideally pcloud support would be even better (it has a music player embedded)

**Comment by jem24h** [comment]
Webdav is transversal which is why we need it, it's supported by :
Cloud storage services: Nextcloud, ownCloud, Box, pCloud, Yandex.Disk, Koofr, Seafile, Mail.ru Cloud, 4shared, DriveHQ, MEGA
Operating systems: Windows, macOS, Linux, iOS, Android
Web servers: Apache HTTP Server, Microsoft IIS, Nginx (with modules)
File managers: Windows Explorer, Finder (macOS), many third-party file managers

**Comment by Kreinoee** [comment]
I suggested looking at an [rclone](https://rclone.org/) based music provider in another feature request for a general clouad storage solution here: https://github.com/orgs/music-assistant/discussions/4470#discussioncomment-15390548


Rclone also support WebDav, so if it is a possibility to use make a rclone based music provider, then WebDav should be supported automatically by that.

**Comment by OzGav** [comment]
WebDAV support will be available in 2.9.0 beta 2

**Comment by OzGav** [comment]
The feature request has been implemented successfully. As such this feature request will be closed and locked. If there are enhancements desired then open a new [feature request](https://github.com/orgs/music-assistant/discussions/categories/feature-requests-and-ideas). If there is a question about how this works then check the [documentation](https://beta.music-assistant.io/), or ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord. If there are problems then ask a question in the [Q&A](https://github.com/orgs/music-assistant/discussions/categories/q-a) area or on Discord or open an [issue](https://github.com/music-assistant/support/issues). Thanks for your interest in seeing this idea being brought to life.
